### PR TITLE
Adds a new User Command

### PR DESCRIPTION
**Pull request by MichaelWDanko**
Discord doesn't let you use the same command as both a SlashCommand and a UserCommand.

To get around this, I created two separate commands and have split out the functions to look up CSR, look up Gamertag, and build the RankedEmbeds that get replied back to the user.

The following tests have been made against a local copy of `hft-backend`:
* Getting a GT's CSR with the `/csr` command
* Getting a Discord user's CSR with the new User command `Get User's CSR` where Discord user has a linked GT
* Getting a Discord user's CSR with the new User command `Get User's CSR` where Discord user does not have a linked GT

### CODE REVIEW

**Comment by MichaelWDanko** [comment]
@DylanBohlender This [commit](https://github.com/HaloFunTime/hft-intern/pull/3/commits/bf9a4dcb5594e223d0179b110202779a81a00c59) was backed out with the last commit since I already created a PR the other day for you to review/consider if you need it.

**Review by DylanBohlender — COMMENTED** [review]
I just approved the CI job so it would run, and it complained about the formatting - mind running the `dev-format.sh` script over everything in your commits so it autoformats everything?

File: commands/csr-slash.js [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+		.addStringOption((option) =>
+		option
+		  .setName("gamertag")
+		  .setDescription("A valid Xbox Live gamertag")
+		  .setMaxLength(15)
+		  .setRequired(true)
+	  	),
+	async execute(interaction) {
+		try {
+			// console.log(`csr-slash.js - About to set const: gamertag`)
```
Comment (DylanBohlender) [inline commands/csr-slash.js]: Generally we want to wipe out any commented debug code before merging things in.

File: commands/csr-slash.js [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+			const gamertag = interaction.options.getString('gamertag');
+			console.log(`csr-slash.js - New value of const gamertag is:`)
+			console.log(gamertag)
```
Comment (DylanBohlender) [inline commands/csr-slash.js]: Same goes for `console.log` statements - mind removing these and the others sprinkled throughout? If they're not necessary in prod they'll use up our log space.

**Comment by DylanBohlender** [comment]
Overall I dig this! Moving the functions to a utils folder makes a lot of sense. There may be some additional file-moving necessary to get the commands properly segmented for deploying, but we can cross that bridge once I'm able to run this locally to test.

Could you push up these commits on a branch in this repo instead of pulling them in from a forked repo?

File: commands/csr-slash.js (hunk 2 above)
Comment (MichaelWDanko) [reply commands/csr-slash.js]: Sure thing, I'll get these unnecessary logs and clean up the code that is commented.

**Comment by MichaelWDanko** [comment]
@DylanBohlender Since I don't have permissions for this repo to create a new branch, I've pushed my changes back up to my forked branch. I also have commits staged for this repo on a new branch that I can push up locally if you'd like but I will first need permissions to create that new branch. 

The files were copied over though after the formatting changes so you should hopefully be able to pass the formatting checks now.

**Comment by MichaelWDanko** [comment]
Closing because this has been reopened in https://github.com/HaloFunTime/hft-intern/pull/4

**Pull request closed without merging**